### PR TITLE
Added logging to X11Grabber

### PR DESCRIPTION
**Pull request by XfableX**
**1.** Tell us something about your changes.
I have added logging to X11Grabber files and the AmlogicGrabber files
**2.** If this changes affect the .conf file. Please provide the changed section

**3.** Reference a issue (optional)
#25 
Note: For further discussions use our forum: forum.hyperion-project.org

### CODE REVIEW

**Comment by penfold42** [comment]
Anyone object to me removing gpio2spi from the build ?

From memory it's not needed anymore and doesn't add to a test repertoire

**Comment by brindosch** [comment]
```
/home/travis/build/hyperion-project/hyperion.ng/src/hyperiond/hyperiond.cpp: In member function ‘void HyperionDaemon::loadConfig(const string&)’:
/home/travis/build/hyperion-project/hyperion.ng/src/hyperiond/hyperiond.cpp:112:29: error: ‘assert’ was not declared in this scope
  assert(schemaData.isValid());
                             ^
make[2]: *** [src/hyperiond/CMakeFiles/hyperiond.dir/hyperiond.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [src/hyperiond/CMakeFiles/hyperiond.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by penfold42** [comment]
Don't merge this - there are compile errors in the amlogic grabber.

Xfablex - it looks like your build environment never built the amlogic grabber so you never saw the errors yourself.

**Comment by XfableX** [comment]
Fixed!

**Comment by redPanther** [comment]
can somebody add
#include <cassert>
in hyperiond.cpp

should fix compile error

**Comment by XfableX** [comment]
I will add it to this request in about an hour

**Comment by redPanther** [comment]
i added the include ....

**Comment by XfableX** [comment]
Ok cool!

**Comment by redPanther** [comment]
and worked pr #73

**Comment by brindosch** [comment]
So this should be moved to logger instance x11 for x11 and instance amlgrabber for amlgrabber? And the AMLGRABBER INFO and AMLGRABBER ERROR and so on should be removed.
https://github.com/hyperion-project/hyperion.ng/issues/25#issuecomment-229317027

File: libsrc/grabber/amlogic/AmlogicGrabber.cpp [hunk 1]
```diff
@@ -30,7 +30,7 @@ AmlogicGrabber::AmlogicGrabber(const unsigned width, const unsigned height) :
 	_height(std::max(160u, height)),
 	_amlogicCaptureDev(-1)
 {
-	std::cout << "AMLOGICGRABBER INFO: [" << __PRETTY_FUNCTION__ << "] constructed(" << _width << "x" << _height << ")" << std::endl;
+	Info(_log, "AMLOGICGRABBER INFO: [%s] constructed(%s x %s)",__PRETTY_FUNCTION__,_width,_height);
```
Comment (penfold42) [inline libsrc/grabber/amlogic/AmlogicGrabber.cpp]: please remove  AMLOGICGRABBER INFO

File: libsrc/grabber/amlogic/AmlogicGrabber.cpp [hunk 2]
```diff
@@ -121,7 +121,7 @@ int AmlogicGrabber::grabFrame(Image<ColorBgr> & image)
 		ioctl(_amlogicCaptureDev, AMVIDEOCAP_IOW_SET_WANTFRAME_HEIGHT, _height) == -1)
 	{
 		// Failed to configure frame width
-		std::cerr << "AMLOGICGRABBER ERROR: [" << __PRETTY_FUNCTION__ << "] Failed to configure capture size (" << errno << "): " << strerror(errno) << std::endl;
+		Error(_log,"AMLOGICGRABBER ERROR: [%s] Failed to configure capture size (%s)",__PRETTY_FUNCTION__,errno,strerror(errno));
```
Comment (penfold42) [inline libsrc/grabber/amlogic/AmlogicGrabber.cpp]: please remove AMLOGICGRABBER ERROR:

File: libsrc/grabber/amlogic/AmlogicGrabber.cpp [hunk 3]
```diff
@@ -134,15 +134,15 @@ int AmlogicGrabber::grabFrame(Image<ColorBgr> & image)
 	const ssize_t bytesRead   = pread(_amlogicCaptureDev, image_ptr, bytesToRead, 0);
 	if (bytesRead == -1)
 	{
-		std::cerr << "AMLOGICGRABBER ERROR: [" << __PRETTY_FUNCTION__ << "] Read of device failed (erno=" << errno << "): " << strerror(errno) << std::endl;
+		Error(_log,"AMLOGICGRABBER ERROR: [%s] Read of device failed (erno=%s): %s",__PRETTY_FUNCTION__,errno,strerror(errno));
```
Comment (penfold42) [inline libsrc/grabber/amlogic/AmlogicGrabber.cpp]: and here

File: libsrc/grabber/amlogic/AmlogicGrabber.cpp [hunk 4]
```diff
@@ -134,15 +134,15 @@ int AmlogicGrabber::grabFrame(Image<ColorBgr> & image)
...
 		close(_amlogicCaptureDev);
 		_amlogicCaptureDev = -1;
 		return -1;
 	}
 	else if (bytesToRead != bytesRead)
 	{
 		// Read of snapshot failed
-		std::cerr << "AMLOGICGRABBER ERROR: [" << __PRETTY_FUNCTION__ << "] Capture failed to grab entire image [bytesToRead(" << bytesToRead << ") != bytesRead(" << bytesRead << ")]" << std::endl;
+		Error(_log,"AMLOGICGRABBER ERROR: [%s] Capture failed to grab entire image [bytesToRead(%s) != bytesRead(%s)]",__PRETTY_FUNCTION__,bytesToRead,bytesRead);
```
Comment (penfold42) [inline libsrc/grabber/amlogic/AmlogicGrabber.cpp]: and here

**Comment by penfold42** [comment]
agreed - @XfableX please remove the unnecessary AMLOGICGRABBER ERROR: from the log messages

**Comment by XfableX** [comment]
done.

**Comment by brindosch** [comment]
faster than light, thank you.

**Pull request merged**